File: binary_search.c
#include<stdio.h>
void binary_search(int n,int arr[n]);
void main(){
    int n;
    printf("enter size of array :");
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
     
    binary_search(n,arr);
}

void binary_search(int n,int arr[n]){
int low=0;
int high=n-1;
int mid;
int target=5;//manual taking we can take from user for better experience
int flag=0;
while(low<=high){
mid=(low+high)/2;
if(arr[mid]==target){
    printf("element found\n");
    flag=1;
    break;
}
else if(arr[mid]>target){
high=mid-1;
}
else if(arr[mid]<target){
    low=mid+1;
}
}
if(flag==0){
    printf("element not found");
}
}

File: Datastructures/BinarySearch.java
public class BinarySearch {
    static void search(int[] arr,int target){
          int low,high,mid;
          boolean flag=false;
          low=0;
          high=arr.length-1;
          while(low<=high){
          mid=(low+high)/2;
          if(arr[mid]==target){
            System.out.println("element found at index :"+mid);
            flag=true;
            break;
          }
          else if(arr[mid]<target){
            low=mid+1;
          }
          else if(arr[mid]>target){
            high=mid-1;
          }
          }
          if(!flag){
            System.out.println("element not found");
        }
    }
    public static void main(String args[]){
      int[] arr={11,22,33,44,55,66,77,88,99};
        search(arr, 77);
    }
}
